xcm-simulator/src/parachain.rs

```diff
@@ -0,0 +1,764 @@
+	type RuntimeOrigin = RuntimeOrigin;
+	type RuntimeCall = RuntimeCall;
+	const VERSION_DISCOVERY_QUEUE_SIZE: u32 = 100;
+	type AdvertisedXcmVersion = pallet_xcm::CurrentXcmVersion;
+}
+
+use thea_primitives::{AuthorityId, AuthoritySignature, Network};
+
+parameter_types! {
+	pub const TheaMaxAuthorities: u32 = 10;
+}
+
+pub struct Executor;
+
+impl thea_primitives::TheaOutgoingExecutor for Executor {
+	fn execute_withdrawals(_network: Network, _withdrawals: Vec<u8>) -> DispatchResult {
+		Ok(())
+	}
+}
+
+impl thea_message_handler::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type TheaId = AuthorityId;
+	type Signature = AuthoritySignature;
+	type MaxAuthorities = TheaMaxAuthorities;
+	type Executor = XcmHelper;
+}
+
+parameter_types! {
+	pub const WithdrawalExecutionBlockDiff: u32 = 7000;
+	pub const XcmHandlerId: PalletId = PalletId(*b"XcmHandl");
+	pub ParachainId: u32 = MsgQueue::parachain_id().into();
+	pub const ParachainNetworkId: u8 = 1;
+	pub NativeAssetId: u128 = 100;
+}
+
+impl xcm_helper::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type AccountIdConvert = LocationToAccountId;
+	type AssetManager = AssetHandler;
+	type AssetCreateUpdateOrigin = EnsureRoot<AccountId>;
+	type Executor = Executor;
+	type AssetHandlerPalletId = AssetHandlerPalletId;
+	type WithdrawalExecutionBlockDiff = WithdrawalExecutionBlockDiff;
+	type ParachainId = ParachainId;
+	type SubstrateNetworkId = ParachainNetworkId;
+	type NativeAssetId = NativeAssetId;
+	type WeightInfo = xcm_helper::weights::WeightInfo<Runtime>;
+}
+
+parameter_types! {
+	pub const AssetDeposit: Balance = 100;
+	pub const ApprovalDeposit: Balance = 1;
+	pub const StringLimit: u32 = 50;
+	pub const MetadataDepositBase: Balance = 10;
+	pub const MetadataDepositPerByte: Balance = 1;
+}
+
+impl pallet_assets::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Balance = Balance;
+	type RemoveItemsLimit = ConstU32<1000>;
+	type AssetId = u128;
+	type AssetIdParameter = codec::Compact<u128>;
+	type Currency = Balances;
+	type CreateOrigin = AsEnsureOriginWithArg<EnsureSigned<AccountId>>;
+	type ForceOrigin = EnsureRoot<AccountId>;
+	type AssetDeposit = AssetDeposit;
+	type AssetAccountDeposit = AssetDeposit;
+	type MetadataDepositBase = MetadataDepositBase;
+	type MetadataDepositPerByte = MetadataDepositPerByte;
+	type ApprovalDeposit = ApprovalDeposit;
+	type StringLimit = StringLimit;
+	type Freezer = ();
+	type Extra = ();
+	type CallbackHandle = ();
+	type WeightInfo = ();
+}
+
+pub struct AccountIdToMultiLocation;
+impl Convert<AccountId, MultiLocation> for AccountIdToMultiLocation {
+	fn convert(account: AccountId) -> MultiLocation {
+		X1(Junction::AccountId32 { network: NetworkId::Any, id: account.into() }).into()
+	}
+}
+
+parameter_types! {
+	pub SelfLocation: MultiLocation = MultiLocation::new(1, X1(Parachain(MsgQueue::parachain_id().into())));
+	pub const BaseXcmWeight: XCMWeight = 100_000_000; // TODO: recheck this @zktony
```

Comment:
Still remains @zktony 
([Here](https://github.com/Polkadex-Substrate/parachain/pull/44#discussion_r1193440665) is an initial comment.)

Reply:
@zktony You mentioned this is fine, if so please remove the TODO